python/ray/serve/api.py

```diff
@@ -536,10 +531,10 @@ def run(
         name,
         parameter_group,
         _blocking=_blocking,
-        remove_past_deployments=remove_past_deployments,
     )
 
     if ingress is not None:
+        client._wait_for_deployment_created(ingress.name)
```

Comment:
With the migration to a reconciler patter in application state manager, there is no guarantee that deployments are created after `controller.deploy_application()` returns; it will take an iteration of `update()` to create those deployments. So before we return a handle to the ingress deployment, we need to wait for the ingress deployment to be created.

Reply:
amazing catch

Reply:
@edoakes Yeah, if `_blocking=True` this isn't necessary because `deploy_application` will wait for the application to become running. It is only to fix the case when `_blocking=False`.